prov/sockets/src/sock_cq.c

```diff
@@ -402,14 +402,32 @@ static ssize_t sock_cq_readerr(struct fid_cq *cq, struct fi_cq_err_entry *buf,
 {
 	struct sock_cq *sock_cq;
 	ssize_t ret;
+	struct fi_cq_err_entry entry;
+	uint32_t api_version;
+	size_t err_data_size = 0;
+	void *err_data = NULL;
 
 	sock_cq = container_of(cq, struct sock_cq, cq_fid);
 	if (sock_cq->domain->progress_mode == FI_PROGRESS_MANUAL)
 		sock_cq_progress(sock_cq);
 
 	fastlock_acquire(&sock_cq->lock);
 	if (ofi_rbused(&sock_cq->cqerr_rb) >= sizeof(struct fi_cq_err_entry)) {
-		ofi_rbread(&sock_cq->cqerr_rb, buf, sizeof(*buf));
+		api_version = sock_cq->domain->fab->fab_fid.api_version;
+		ofi_rbread(&sock_cq->cqerr_rb, &entry, sizeof(entry));
+
+		if ((FI_VERSION_GE(api_version, FI_VERSION(1, 5)))
+			&& buf->err_data && buf->err_data_size) {
+			err_data = entry.err_data;
+			err_data_size = MIN(entry.err_data_size, buf->err_data_size);
+		}
+
+		*buf = entry;
+		if (err_data) {
+			memcpy(buf->err_data, entry.err_data, err_data_size);
+			buf->err_data_size = err_data_size;
+		}
```

Comment:
Thank you! This is pretty nice point ;)
Sorry that with some delay.
New path set have been prepared.